gtsam/geometry/Pose3.cpp

```diff
@@ -63,6 +63,92 @@ Matrix6 Pose3::AdjointMap() const {
   return adj;
 }
 
+/* ************************************************************************* */
+// Calculate AdjointMap applied to xi_b, with Jacobians
+Vector6 Pose3::Adjoint(const Vector6& xi_b, OptionalJacobian<6, 6> H_this,
+                       OptionalJacobian<6, 6> H_xib) const {
+  //  Ad   *   xi   =   [   R   0    .  [w
+  //                      [t]R  R ]      v]
+  // Declarations, aliases, and intermediate Jacobians easy to compute now
+  Vector6 result;
+  auto Rw = result.head<3>();
```

Comment:
This is initialized at this point

Reply:
Can you clarify what you mean in this comment?

`Rw` is meant to be an "output variable" which is first assigned `Rw = R_.rotate(w, ...` then later used.  This is to avoid creating an extra Vector3 and then later copying it over to the result.

Reply:
That shouldn't be a problem since Eigen's templating logic + the C++ compiler will optimize this to avoid that extra copy. We should optimize for readability and let the library handle the underlying complexities, especially for simple cases like this. :slightly_smiling_face: